dashboard/app/models/ability.rb

```diff
@@ -318,12 +318,20 @@ def initialize(user)
       end
     end
 
+    # Checks if user is directly enrolled in pilot
     if user.persisted?
       if Experiment.enabled?(user: user, experiment_name: 'csa-pilot')
         can :get_access_token, :javabuilder_session
       end
     end
 
+    # Checks if user has a teacher enrolled in pilot
+    if user.persisted?
+      if user.teachers.any? {|t| t.has_pilot_experiment?('csa-pilot')}
```

Comment:
What happens if the user has no teachers? Will it be an empty array or could it be nil?

Reply:
I noticed in other cases an !.empty check, so added that here to avoid any errors